lib/assertions.js

```diff
@@ -786,7 +786,15 @@ module.exports = (expect) => {
             );
           });
           if (expect.flags.only) {
-            expect(subject.length, 'to be', args.length);
+            expect(
+              subject &&
+                Array.prototype.every.call(subject, (item) =>
+                  Array.prototype.some.call(args, (arg) =>
+                    expect.equal(item, arg)
+                  )
+                ),
+                'to be truthy'
+            );
```

Comment:
> @sunesimonsen:
> You need to run through the RHS items, for each of them you check if it matches each of the items in the LHS. You record that information in an array of boolean, where you always or the results. If the RHS item doesn't match any items you fail. When you are done you check that every item in the array is true; otherwise you fail.
>
> I think the above would work, but I also realise that I can't ask you to go an do that, so I'll probably volunteer to do it. But I guess you need to solution now, so that is a bit of pickle 😅


If I understand your description correctly, I did the almost the opposite here. I run through all the items in the LHS, making sure that each of them equals at least one of the specs on the RHS. That was the simplest way I could come up with on the spot.

Does it have significantly worse performance characteristics than your proposed solution? If so I'm all for taking a stab at inversing it :-)

Reply:
My take: It doesn't only contain the elements `1` and `2` because it doesn't contain `2`. To me, the `only` bit is an extra requirement.

Reply:
@papandreou I opened this thread review thread myself, just as @sunesimonsen opened his in the same spot. I marked this resolved, and I'll repost your comment up in the other one :) Sorry for the mess 🙊